swift/StableDiffusion/pipeline/StableDiffusionPipeline.Configuration.swift

```diff
@@ -44,7 +44,19 @@ public struct PipelineConfiguration: Hashable {
     public var encoderScaleFactor: Float32 = 0.18215
     /// Scale factor to use on the latent before decoding
     public var decoderScaleFactor: Float32 = 0.18215
-    
+    /// If `originalSize` is not the same as `targetSize` the image will appear to be down- or upsampled.
```

Comment:
Yea, I'd agree it probably shouldn't be a requirement on the caller to set the scale factor properly, especially when we already know the pipeline is SDXL and can set the default properly. My question would be whether we'd want a new class for this or just setup the defaults somewhere in the pipeline. Or potentially a broader option of just including the config files with the converted models directly, alongside the merges.txt and vocab.json, and using those for the default config.

Reply:
> potentially a broader option of just including the config files with the converted models directly

That would be nice and we can even make that backward compatible. However, it still wouldn't address the argument set mismatch across XL and non-XL right? IMO, the easiest is to create 2 (XL and non-XL) default configs and then add your proposal from above as an extension.

Reply:
Yea that makes sense to me, and seems to be the approach `Diffusers` likes to take. My only nitpick is the repeated code, be it can be consolidated a fair amount by committing to the two pipelines and pulling out shared code as needed.

Reply:
I agree, we can refactor over time :)

Reply:
I agree with the approach discussed, separate configs do seem necessary to make usage simpler.